Packs/ExabeamSecurityOperationsPlatform/Integrations/ExabeamSecOpsPlatform/ExabeamSecOpsPlatform_test.py

```diff
@@ -0,0 +1,240 @@
+        ]
+    }
+
+    client = MockClient("", "", "", False, False)
+
+    mocker.patch.object(client, "search_request", return_value=mock_response)
+
+    # Define test arguments
+    args = {
+        'query': '',
+        'fields': 'message',
+        'limit': '50',
+        'start_time': '2024-05-01T00:00:00',
+        'end_time': '2024-05-08T00:00:00'
+    }
+
+    # Call the search_command function
+    response = search_command(client, args)
+
+    assert isinstance(response, CommandResults)
+    assert response.outputs_prefix == "ExabeamPlatform.Event"
+    assert response.outputs == mock_response["rows"]
+    expected_readable_output = (
+        "### Logs\n"
+        "|Id|Is Parsed|Raw Log Ids|Raw Logs|Tier|\n"
+        "|---|---|---|---|---|\n"
+        "| 123 | false | 1 | fictive | Tier |\n"
+        "| 456 |  |  |  |  |\n"
+    )
+    assert expected_readable_output in response.readable_output
+
+
+def test_search_command_failure(mocker):
+    """
+    GIVEN:
+        A mocked Exabeam client and invalid search query arguments.
+
+    WHEN:
+        'search_command' function is called with invalid arguments.
+
+    THEN:
+        It should raise a DemistoException.
+    """
+    # Mocking the client to simulate a response with errors
+    client = MockClient("", "", "", False, False)
+    mocker.patch.object(client, "search_request", return_value={"errors": {"message": "Error occurred"}})
+
+    args = {
+        'query': '',
+        'fields': 'message',
+        'limit': '50',
+        'start_time': '2024-05-01T00:00:00',
+        'end_time': '2024-05-08T00:00:00'
+    }
+
+    with pytest.raises(DemistoException, match="Error occurred"):
+        search_command(client, args)
+
+
+def test_get_date(mocker):
+    """
+    GIVEN:
+        a mocked CommonServerPython.arg_to_datetime function returning a specific time string,
+
+    WHEN:
+        'get_date' function is called with the provided time string,
+
+    THEN:
+        it should return the date part of the provided time string in the 'YYYY-MM-DD' format.
+    """
+    time = '2024.05.01T14:00:00'
+    expected_result = '2024-05-01T14:00:00Z'
+
+    with mocker.patch("CommonServerPython.arg_to_datetime", return_value=time):
+        result = get_date(time, "start_time")
+
+    assert result == expected_result
+
+
+@pytest.mark.parametrize('input_str, expected_output', [
+    (
+        "key:Some Value",
+        'key:"Some Value"'
+    ),
+    (
+        "key:TrUe",
+        "key:true"
+    ),
+    (
+        "key:false",
+        "key:false"
+    )
+])
+def test_transform_string(input_str, expected_output):
+    """
+    GIVEN:
+        An input string to be transformed.
+    WHEN:
+        The 'transform_string' function is called with the input string.
+    THEN:
+        It should transform the input string according to the specified rules.
+    """
+    assert transform_string(input_str) == expected_output
+
+
+@pytest.mark.parametrize('input_str, expected_output', [
```

Comment:
Please add a test involving all the scenarios together true/false and a string value.